e2e/template.go

```diff
@@ -28,7 +28,7 @@ func baseTemplate(location string) *datamodel.NodeBootstrappingConfiguration {
 				ProvisioningState: "",
 				OrchestratorProfile: &datamodel.OrchestratorProfile{
 					OrchestratorType:    "Kubernetes",
-					OrchestratorVersion: "1.26.0",
+					OrchestratorVersion: "1.29.6",
```

Comment:
actually, instead of hard-coding this value at all, maybe we can infer this value from the abe2e cluster's control plane properties instead?

@r2k1 maybe something you could look at?

Reply:
agreed to hard-code for now to unblock testing on 1.29

Reply:
What happened? Why do you need to change it? Did test cluster auto-upgraded?

Reply:
I think, ideally instead of testing old/current version we should test the latest version

Reply:
1.26 has been EOL since February: https://kubernetes.io/releases/patch-releases/#non-active-branch-history

I need to update it so I can add E2E tests for this change, which is specific to k8s 1.29 or later: https://github.com/Azure/AgentBaker/pull/4824